src/middleware/validate/validateBooleanMiddleware.ts

```diff
@@ -0,0 +1,53 @@
+// Copyright 2017-2022 Parity Technologies (UK) Ltd.
+// This file is part of Substrate API Sidecar.
+//
+// Substrate API Sidecar is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import { RequestHandler } from 'express';
+import { BadRequest } from 'http-errors';
+
+/**
+ * Validate that the given query params that are expected to be booleans are correct.
+ *
+ * @param queryParams An array of queryParams to check for. These are passed in at the controller level.
+ */
+export const validateBooleanMiddleware = (
+	queryParams: string[]
+): RequestHandler => {
+	return (req, _res, next) => {
+		const queryKeys = Object.keys(req.query);
+		const errQueryParams: string[] = [];
+		queryKeys
+			.filter((queryParam) => queryParams.includes(queryParam))
```

Comment:
Nit; not a big deal but this isn't the most efficient approach (looking over the keys and then for each key looping over the provided params (prolly only one or two).

More efficient would be something like:

```
for (const key of queryParams) {
    const param = req.query[key].toString().toLowerCase();
    if (param === 'true' || param === 'false') {
        // ....
    }
}
```

Ie only check the queryparams we asked to validate and ignore the rest completely :)

Reply:
Ahh yes good catch, will fix that up

Reply:
fixed